src/main/java/org/jenkinsci/plugins/awsdevicefarm/AWSDeviceFarmRecorder.java

```diff
@@ -557,6 +579,46 @@ public void perform(@Nonnull hudson.model.Run<?, ?> build, @Nonnull FilePath wor
             writeToLog(log, String.format("Using Project '%s'", projectName));
             Project project = adf.getProject(projectName);
 
+            // Device Selection logic
+            DeviceSelectionConfiguration deviceSelectionConfig = null;
+            ArrayList<DeviceFilter> deviceFilterList = null;
+            DeviceFilter deviceFilter = null;
+
+            if (deviceSelection != null && !deviceSelection.isEmpty()) {
+                deviceSelectionConfig = new DeviceSelectionConfiguration();
+                deviceFilter = new DeviceFilter();
+                deviceFilterList = new ArrayList<DeviceFilter>();
+
+                String deviceSelectionStr = null;
+                deviceSelectionStr = String.format("%s", env.expand(deviceSelection));
+                String[] deviceSelectionList = deviceSelectionStr.split(",");
```

Comment:
You may want to consider .trim() before .split() deviceFilterString to remove any leading or trailing whitespace, ensuring that parsing is more robust.